airframe-http-recorder/src/main/scala/wvlet/airframe/http/recorder/HttpRecorder.scala

```diff
@@ -99,6 +97,23 @@ object HttpRecorder extends LogSupport {
     new HttpRecorderServer(recorder, finagleConfig, new ReplayService(recorder))
   }
 
+  /**
+    * Creates an HTTP server that returns programmed HTTP responses.
+    * If no matching record is found, use the given fallback handler.
+    */
+  def createProgrammableServer(programmer: HttpRecordStore => Unit): FinagleServer = {
+    val recorderConfig = HttpRecorderConfig("localhost")
+    val finagleConfig = FinagleServerConfig(recorderConfig.serverPort)
+    val recorder      = new HttpRecordStore(recorderConfig, true){
+      override def requestHash(request: Request): Int = {
+        val content = request.getContentString()
+        s"${request.method.toString()}:${recorderConfig.destAddress.hostAndPort}${request.uri}:${content.hashCode}".hashCode
+      }
```

Comment:
It's difficult to program exact same headers. I think matching by method, address, uri and content is good enough in the programmable HTTP mock server.

Reply:
ok. Let's add a customization entry point to HttpRecordStore later to add user-defined pattern matching rules.

No change is required for now.

Reply:
My idea is adding a method like:
```
HttpRecordStore.withRequestMatcher(matcher:PartialFunction[Request, Response])
```